base/steer/FairRootManager.cxx

```diff
@@ -258,25 +220,11 @@ void FairRootManager::RegisterImpl(const char* name, const char *folderName, T*
   }
 
   if(toFile) { /**Write the Object to the Tree*/
-    TFolder* folder=0;
-    TFolder* f=0;
-    if(fOutFolder==0) {
-      f=static_cast<TFolder*>(fRootFolder->FindObjectAny(folderName));
-      if(f==0) {
-        folder= fRootFolder->AddFolder(folderName,folderName);
-      } else {
-        folder=f;
-      }
-    } else {
-      f=static_cast<TFolder*>(fOutFolder->FindObjectAny(folderName));
-      if(f==0) {
-        folder= fOutFolder->AddFolder(folderName,folderName);
-      } else {
-        folder=f;
-      }
-    }
```

Comment:
This looks like a large rewrite of FairRootManager. Is the functionality backward compatible? I guess we need to be sure this works correctly.